src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -1851,6 +1851,10 @@ public function setKeyType($type)
      */
     public function getIncrementing()
     {
+        if ($this->getKeyType() === 'string') {
+            return false;
+        }
+
```

Comment:
This might need to be rethought. In my mind a string key type column can never be incrementing but I might be wrong.